src/rdkafka_metadata.c

```diff
@@ -682,8 +766,14 @@ rd_kafka_parse_Metadata(rd_kafka_broker_t *rkb,
                                 rd_kafka_metadata_cache_topic_update(
                                     rk, &md->topics[i], &mdi->topics[i],
                                     rd_false /*propagate later*/,
-                                    has_client_rack, mdi->brokers,
-                                    md->broker_cnt);
+                                    has_client_rack, /* use has_client_rack
+                                                       rather than
```

Comment:
Comment style, maybe change it to:

> use has_client_rack rather than compute_racks. We need cached rack ids only in case we need to rejoin the group if they change and client.rack is set. As specified in KIP-881.

Reply:
I've made this change, but the GitHub UI makes it a bit difficult to understand the intent (I've changed the text and the position slightly)